W/Wayland/build_tarballs.jl

```diff
@@ -3,29 +3,51 @@
 using BinaryBuilder
 
 name = "Wayland"
-version = v"1.21.0"
+version = v"1.23.0"
 
 # Collection of sources required to build Wayland
 sources = [
     ArchiveSource("https://gitlab.freedesktop.org/wayland/wayland/-/releases/$(version)/downloads/wayland-$(version).tar.xz",
-                  "6dc64d7fc16837a693a51cfdb2e568db538bfdc9f457d4656285bb9594ef11ac"),
+                  "05b3e1574d3e67626b5974f862f36b5b427c7ceeb965cb36a4e6c2d342e45ab2"),
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
 cd $WORKSPACE/srcdir/wayland-*/
 
-ln -s `which wayland-scanner` $bindir
-cp $prefix/libdata/pkgconfig/* $prefix/lib/pkgconfig || true
+# Build a native version of wayland-scanner
 
-mkdir build-wayland
+cat - >host_pkgconfig.meson <<EOF
+[binaries]
+pkgconfig = ['/usr/bin/pkg-config', '--define-variable=prefix=${host_prefix}']
+EOF
 
-cd build-wayland
-meson .. \
+PKG_CONFIG_SYSROOT_DIR='' \
+PKG_CONFIG_PATH="${host_libdir}/pkgconfig" \
+meson setup host_build . \
```

Comment:
> and at the same time, try to resolve a bootstrap issue with `wayland-scanner` being a build-time dependency of building the other libraries.

It isn't clear to me why we shouldn't be using `HostBuildDependency` as it was done before.

Reply:
Trying to upgrade to v1.23 would only see the v1.21 version of `wayland-scanner`, and the build system would complain and abort. Trying to build the native & target from the same source was my way to resolve that issue.

Reply:
Explicitly, the error that occurs if only the version & hash are changed and a rebuild is attempted is:
```
...
[14:20:04] Build-time dependency wayland-scanner found: NO (tried cmake)
[14:20:04] 
[14:20:04] ../src/meson.build:84:15: ERROR: Dependency lookup for wayland-scanner with method 'pkgconfig' failed: Invalid version, need 'wayland-scanner' ['1.23.0'] found '1.21.0'.
...
```